src/OrleansGetEventStore/GetEventStoreProvider.cs

```diff
@@ -0,0 +1,343 @@
+        }
+
+        public const string ConnectionStringParameterName = "ConnectionString";
+        public const string PageSizeParameterName = "PageSize";
+        public const string StoreAllTypenamesParameterName = "StoreAllTypenames";
+        public const string StoreObjectIdentityParameterName = "StoreObjectIdentity";
+
+        public string DefaultStreamName(Type grainType, GrainReference grainReference)
+        {
+            return string.Format("{0}_{1}", serviceId, grainReference.ToKeyString());
+        }
+
+        // default server configuration used by GetEventStore, we match them for easy testing on a local machine
+        public const string GetEventStoreDefaultConnectionString = "tcp://admin:changeit@localhost:1113";
+
+        private static readonly byte[] emptyMetaData = new byte[0];
+
+        public async Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            serviceId = providerRuntime.ServiceId.ToString();
+
+            this.Name = name;
+            this.Log = providerRuntime.GetLogger(string.Format("LogViews.{0}.{1}", GetType().Name, id));
+
+            pageSize = config.GetIntProperty(PageSizeParameterName, 500);
+            var storeAllTypenames = config.GetBoolProperty(StoreAllTypenamesParameterName, false);
+            var storeObjectIdentity = config.GetBoolProperty(StoreObjectIdentityParameterName, false);
+
+            var connectionString = config.GetProperty(ConnectionStringParameterName, GetEventStoreDefaultConnectionString);
+            Log.Info($"Init Name={Name} ConnectionString={connectionString} PageSize={pageSize} StoreAllTypenames={storeAllTypenames} StoreObjectIdentity={storeObjectIdentity} Severity={Log.SeverityLevel}");
+
+            var serializationManager = providerRuntime.ServiceProvider.GetService<SerializationManager>();
+            jsonSettings = OrleansJsonSerializer.UpdateSerializerSettings(OrleansJsonSerializer.GetDefaultSerializerSettings(serializationManager, providerRuntime.GrainFactory), config);
+            jsonSettings.TypeNameHandling = storeAllTypenames ? TypeNameHandling.All : TypeNameHandling.Auto;
+            jsonSettings.PreserveReferencesHandling = storeObjectIdentity ? PreserveReferencesHandling.Objects : PreserveReferencesHandling.None;
+
+            // connect to the GetEventStore service
+            var uri = new Uri(connectionString);
+            var settings = ConnectionSettings.Create().KeepReconnecting().KeepRetrying();
+            this.connection = EventStoreConnection.Create(settings, uri);
+            await this.connection.ConnectAsync().ConfigureAwait(false);
+
+            Log.Verbose("Successfully Connected");
+        }
+
+        public Task Close()
+        {
+            this.connection.Close();
+            return Task.CompletedTask;
+        }
+
+
+        public async Task<bool> Append<E>(string streamName, IEnumerable<KeyValuePair<Guid,E>> events, int? expectedVersion)
```

Comment:
All public/protected members of public types should have XML comments (it's fine to use `/// <inheritdoc />` if it should inherit the doc from the parent interface or class)